libs/coin-modules/coin-ton/src/bridge/bridgeHelpers/txn.ts

```diff
@@ -164,3 +197,109 @@ export function mapTxToOps(
     return ops;
   };
 }
+
+export function mapJettonTxToOps(
+  accountId: string,
+  addr: string,
+  addressBook: TonAddressBook,
+): (tx: TonJettonTransfer) => TonOperation[] {
+  return (tx: TonJettonTransfer): TonOperation[] => {
+    const accountAddr = Address.parse(addr).toString({ urlSafe: true, bounceable: false });
+    if (accountAddr !== addr) throw Error(`[ton] unexpected address ${accountAddr} ${addr}`);
+
+    const jettonMasterAddr = Address.parse(tx.jetton_master).toString({
+      urlSafe: true,
+      bounceable: true,
+    });
+    const tokenCurrency = findTokenByAddressInCurrency(
+      jettonMasterAddr.toLowerCase(),
+      decodeAccountId(accountId).currencyId,
+    );
+    if (!tokenCurrency) return [];
+    const tokenAccountId = encodeTokenAccountId(accountId, tokenCurrency);
+
+    const ops: TonOperation[] = [];
+    const isReceiving = addressesAreEqual(
+      accountAddr,
+      Address.parse(tx.destination).toString({ urlSafe: true, bounceable: false }),
+    );
+    const isSending = addressesAreEqual(
+      accountAddr,
+      Address.parse(tx.source).toString({ urlSafe: true, bounceable: false }),
+    );
+    if (!isSending && !isReceiving) throw Error("[ton] unexpected addresses");
+
+    const date = new Date(tx.transaction_now * 1000); // now is defined in seconds
+    const hash = tx.transaction_hash;
+
+    if (isReceiving) {
+      ops.push({
+        id: encodeOperationId(tokenAccountId, hash, "IN"),
+        hash,
+        type: "IN",
+        value: BigNumber(tx.amount),
+        fee: BigNumber(0),
+        blockHeight: 1, // we don't have block info
+        blockHash: null,
+        hasFailed: false,
+        accountId: tokenAccountId,
+        senders: getFriendlyAddress(addressBook, tx.source),
+        recipients: [accountAddr],
+        date,
+        extra: {
+          lt: tx.transaction_lt,
+          explorerHash: hash,
+          comment: {
+            isEncrypted: false,
+            text: tx.forward_payload ? decodeForwardPayload(tx.forward_payload) : "",
+          },
+        },
+      });
+    }
+
+    if (isSending) {
```

Comment:
I know it is not possible to send a tx to the itself in Ledger live. but if the account is exported from other wallets, a transaction is interpreted as two operations if the recipient and the sender is the same?

Reply:
Yes, in this case we would receive one only transaction from the endpoint (for the IN and OUT) and we would display both (eceived and sent transactio would be displayed).